app/services/live_stream_service.rb

```diff
@@ -0,0 +1,123 @@
+require 'google/api_client'
+require 'google/api_client/client_secrets'
+require 'google/api_client/auth/file_storage'
+require 'google/api_client/auth/installed_app'
+require 'time'
+
+module LiveStreamService
+  # This OAuth 2.0 access scope allows for read-only access to the authenticated
+  # user's account, but not other types of account access.
+  YOUTUBE_READONLY_SCOPE = 'https://www.googleapis.com/auth/youtube'
+  YOUTUBE_API_SERVICE_NAME = 'youtube'
+  YOUTUBE_API_VERSION = 'v3'
+  
+  def self.get_authenticated_service
+    client = Google::APIClient.new(
+      :application_name => $PROGRAM_NAME,
+      :application_version => '1.0.0'
+    )
+    youtube = client.discovered_api(YOUTUBE_API_SERVICE_NAME, YOUTUBE_API_VERSION)
+  
+    file_storage = Google::APIClient::FileStorage.new("#{$PROGRAM_NAME}-oauth2.json")
```

Comment:
Can this be done using our standard method of having the api_key in environment variables?  Having this 'oauth2.json' file is a different method and would take administrative overhead to keep it working.